ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.tests
+
+import java.time.{Duration, Instant}
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.testtool.infrastructure.{
+  Allocation,
+  Assertions,
+  LedgerApiServer,
+  LedgerSession,
+  LedgerTestSuite
+}
+import com.digitalasset.ledger.api.v1.command_completion_service.{
+  CompletionEndRequest,
+  CompletionStreamRequest,
+  CompletionStreamResponse
+}
+import com.digitalasset.ledger.api.v1.command_submission_service.SubmitRequest
+import com.digitalasset.ledger.client.binding.{Primitive => P}
+import com.digitalasset.ledger.api.v1.commands.{Command, Commands}
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.transaction.Transaction
+import com.digitalasset.ledger.api.v1.transaction_filter.{Filters, TransactionFilter}
+import com.digitalasset.ledger.api.v1.transaction_service.{
+  GetTransactionsRequest,
+  GetTransactionsResponse
+}
+import io.grpc.{Context, Status}
+import io.grpc.stub.StreamObserver
+import scalaz.syntax.tag._
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.{ExecutionContext, Future, Promise, blocking}
+import scala.util.{Failure, Success, Try}
+import com.digitalasset.ledger.test_dev.{PingPong => M}
+import org.slf4j.Logger
+
+case class Envelope(name: String, transactionSizeKb: Int, throughput: Int, latencyMs: Int)
+
+object Envelope {
+
+  /** test will unlikely fail */
+  object ProofOfConcept
+      extends Envelope("PoC", transactionSizeKb = 1, throughput = 0, latencyMs = 60000)
+
+  /** test will fail if performance is lower than alpha envelope */
+  object Alpha extends Envelope("Alpha", transactionSizeKb = 100, throughput = 5, latencyMs = 3000)
+
+  /** test will fail if performance is lower then beta envelope */
+  object Beta extends Envelope("Beta", transactionSizeKb = 1000, throughput = 20, latencyMs = 1000)
+
+  object Public
+      extends Envelope("Public", transactionSizeKb = 5000, throughput = 50, latencyMs = 1000)
+
+  object Enterprise
+      extends Envelope("Enterprise", transactionSizeKb = 25000, throughput = 500, latencyMs = 500)
+
+}
+
+trait PerformanceEnvelope {
```

Comment:
Perhaps let's rename this to `PerformanceTests`.

Reply:
The tests (benchmarks, really) pass only relative to a performance envelope though, which we might want to actually keep. What do you think @meiersi-da?